### PR TITLE
TELCODOCS#1912: Added API endpoints for delete subscription and delete all subscriptions

### PR DESCRIPTION
**Pull request by sr1kar99**
Version(s):
4.12+

Issue:
[TELCODOCS-1912](https://issues.redhat.com/browse/TELCODOCS-1912)

Link to docs preview:

- [Subscribing DU applications to PTP events with the REST API](https://78920--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ptp/using-ptp-events#subscribing-du-applications-to-ptp-events-rest-api-reference_using-ptp-hardware-fast-events-framework)
- [PTP events REST API reference](https://78920--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ptp/using-ptp-events#cnf-fast-event-notifications-api-refererence_using-ptp-hardware-fast-events-framework)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by sr1kar99** [comment]
Hi @aneeshkp 
Could you please review this PR when you get a chance?
Also, could you share the TBD information?

Thanks!

File: modules/cnf-fast-event-notifications-api-refererence.adoc [hunk 1]
```diff
@@ -97,6 +113,32 @@ Returns details for the subscription with ID `<subscription_id>`
...
+|Type
+
+|`<subscription_id>`
+|string
+|====
+
+.Example API response
+[source,json]
+----
+TBD
```
Comment (jzding) [inline modules/cnf-fast-event-notifications-api-refererence.adoc:139]: {
    "status": "OK"
}

File: modules/cnf-fast-event-notifications-api-refererence.adoc [hunk 2]
```diff
@@ -63,6 +63,22 @@ Creates a new subscription. If a subscription is successfully created, or if it
...
+
+[discrete]
+=== Description
+
+Deletes all subscriptions.
+
+.Example API response
+[source,json]
+----
+TBD
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/cnf-fast-event-notifications-api-refererence.adoc]: Added this info!

File: modules/cnf-fast-event-notifications-api-refererence.adoc (hunk 1 above)
Comment (sr1kar99) [reply modules/cnf-fast-event-notifications-api-refererence.adoc]: Added this info!

**Review by jzding — APPROVED** [review]
/lgtm

File: modules/cnf-fast-event-notifications-api-refererence.adoc [hunk 3]
```diff
@@ -97,6 +115,34 @@ Returns details for the subscription with ID `<subscription_id>`
...
+
+Deletes the subscription with ID `<subscription_id>`.
+
+.Query parameters
+[cols=2*, width="60%", options="header"]
+|====
+|Parameter
+|Type
+
+|`<subscription_id>`
```
Comment (dfitzmau) [inline modules/cnf-fast-event-notifications-api-refererence.adoc:134]: Is `<subscription_id>` actually the parameter name? Usually parameter names are exact and not a user-replaced value.Maybe the table could be removed and the string type could be added as a sentence?

**Review by dfitzmau — APPROVED** [review]
/remove-label peer-review-in-progress

/label peer-review-done

File: modules/cnf-fast-event-notifications-api-refererence.adoc (hunk 3 above)
Comment (sr1kar99) [reply modules/cnf-fast-event-notifications-api-refererence.adoc:134]: Hi @jzding 

I noticed in one of the API reference doc sections that parameters as part of the API endpoints have been called "Global path parameters". 
Link - https://78920--ocpdocs-pr.netlify.app/openshift-enterprise/latest/rest_api/console_apis/consoleclidownload-console-openshift-io-v1#apisconsole-openshift-iov1consoleclidownloadsname

So can we update this API end point as follows:
"DELETE api/ocloudNotifications/v1/subscriptions/{subscription_id}`"

And change the table name from "Query parameters" to "Global path parameters"?

Comment (jzding) [reply modules/cnf-fast-event-notifications-api-refererence.adoc:134]: yes {subscription_id} should be Global path parameters

Comment (jzding) [reply modules/cnf-fast-event-notifications-api-refererence.adoc:134]: same applies to `resource_address` in api/ocloudNotifications/v1/<resource_address>/CurrentState

Comment (sr1kar99) [reply modules/cnf-fast-event-notifications-api-refererence.adoc]: Hi @dfitzmau , @jzding 
Updated as per the discussion above.
Thanks!

**Comment by kcarmichael08** [comment]
Sorry, @sr1kar99 , it looks like you will need to create a manual cherry pick for 4.12 and 4.13.

**Pull request merged**